doc/releases/changelog-dev.md

```diff
@@ -68,6 +68,28 @@
 
 <h4>Always differentiable 📈</h4>
 
+* The Hadamard test gradient tranform `qml.gradients.hadamard_grad` is now available.
+  [#3625](https://github.com/PennyLaneAI/pennylane/pull/3625)
+
+  This hardware compatible transform allows you to get the gradient of your circuit by
+  performing a Hadamard test on your circuit where one added the controlled generator 
+  of the trainable unitary. Your device needs an auxiliary wire to perform it.
```

Comment:
I'm not 100% sure I understand

> where one added the controlled generator of the trainable unitary.

That said, usually if there are arguments to a function that are vaguely nuanced, we leave them out of the changelog — keen users will go to the docstring anyway!

```suggestion
* The Hadamard test gradient tranform is now available via`qml.gradients.hadamard_grad`.
  [#3625](https://github.com/PennyLaneAI/pennylane/pull/3625)

  `qml.gradients.hadamard_grad` is a hardware-compatible transform that calculates the 
  gradient of a quantum circuit using the Hadamard test. Note that the device requires an 
  auxiliary wire to calculate the gradient.
```

I would also explain in the code example provided below which wire it is that you're saying is the "auxiliary" wire — maybe give it a verbose name like `"auxiliary"`?